src/components/Link.react.js

```diff
@@ -27,28 +26,32 @@ export default class Link extends Component {
         this.updateLocation = this.updateLocation.bind(this);
     }
 
-    updateLocation() {
-        const {href, refresh} = this.props;
+    updateLocation(e) {
+        e.preventDefault(); // prevent anchor from updating location
+        const { href, refresh } = this.props;
         if (refresh) {
             window.location.pathname = href;
         } else {
             window.history.pushState({}, '', href);
             window.dispatchEvent(new CustomEvent('onpushstate'));
         }
+        window.scrollTo(0, 0); // scroll back to top
```

Comment:
Very sweet. For some reason I thought we would need to do this _after_ the content was updated, but after playing around with it seems like this totally works 👏